include/nbl/builtin/hlsl/workgroup/shared_arithmetic.hlsl

```diff
@@ -0,0 +1,9 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_WORKGROUP_SHARED_ARITHMETIC_INCLUDED_
+#define _NBL_BUILTIN_HLSL_WORKGROUP_SHARED_ARITHMETIC_INCLUDED_
+
+#include "nbl/builtin/hlsl/workgroup/shared_clustered.hlsl"
+
+#endif
```

Comment:
LLVM/Lavapipe have subgroup sizes of 4, ARM/Intel can do 16 sometimes, this means that your first prefix sum needs `((_NBL_HLSL_WORKGROUP_SIZE_-1)>>2)+1` shared memory slots

Reply:
nevermind everyone has a shuffle now!

Reply:
but please nuke the file and any other unused thing!

Reply:
when sweeping up you store between `N/MAX_SUBGROUP_SIZE` and `N/MIN_SUBGROUP_SIZE` elements on the first sweep and same amount on the penultimate downsweep

if you want to avoid having 2x as many barriers (so reading from scratch and writing to scratch at the same time), then you need `((_NBL_HLSL_WORKGROUP_SIZE_-1)>>MIN_SUBGROUP_SIZE_LOG2)+((_NBL_HLSL_WORKGROUP_SIZE_-1)>>(2*MIN_SUBGROUP_SIZE_LOG2))+2` slots alltogether